File: mediamtx-camera-service/sdk/python/mediamtx_camera_sdk/exceptions.py
"""
Custom exceptions for MediaMTX Camera Service SDK.

This module contains the exception classes used by the SDK for error handling.
"""


class CameraServiceError(Exception):
    """Base exception for camera service errors."""
    pass


class AuthenticationError(CameraServiceError):
    """Authentication failed."""
    pass


class ConnectionError(CameraServiceError):
    """Connection failed."""
    pass


class CameraNotFoundError(CameraServiceError):
    """Camera device not found."""
    pass


class MediaMTXError(CameraServiceError):
    """MediaMTX operation failed."""
    pass


class TimeoutError(CameraServiceError):
    """Operation timed out."""
    pass


class ValidationError(CameraServiceError):
    """Input validation failed."""
    pass

File: mediamtx-camera-service/sdk/javascript/src/exceptions.ts
/**
 * Custom exceptions for MediaMTX Camera Service SDK.
 */

export class CameraServiceError extends Error {
    constructor(message: string) {
        super(message);
        this.name = 'CameraServiceError';
    }
}

export class AuthenticationError extends CameraServiceError {
    constructor(message: string) {
        super(message);
        this.name = 'AuthenticationError';
    }
}

export class ConnectionError extends CameraServiceError {
    constructor(message: string) {
        super(message);
        this.name = 'ConnectionError';
    }
}

export class CameraNotFoundError extends CameraServiceError {
    constructor(message: string) {
        super(message);
        this.name = 'CameraNotFoundError';
    }
}

export class MediaMTXError extends CameraServiceError {
    constructor(message: string) {
        super(message);
        this.name = 'MediaMTXError';
    }
}

export class TimeoutError extends CameraServiceError {
    constructor(message: string) {
        super(message);
        this.name = 'TimeoutError';
    }
}

export class ValidationError extends CameraServiceError {
    constructor(message: string) {
        super(message);
        this.name = 'ValidationError';
    }
}
